frontend/src/scenes/surveys/surveysLogic.tsx

```diff
@@ -48,22 +53,95 @@ export const surveysLogic = kea<surveysLogicType>([
         setSearchTerm: (searchTerm: string) => ({ searchTerm }),
         setSurveysFilters: (filters: Partial<SurveysFilters>, replace?: boolean) => ({ filters, replace }),
         setTab: (tab: SurveysTabs) => ({ tab }),
+        loadNextPage: true,
+        loadNextSearchPage: true,
     }),
     loaders(({ values }) => ({
-        surveys: {
-            __default: [] as Survey[],
+        data: {
+            __default: {
+                surveys: [] as Survey[],
+                surveysCount: 0,
+                searchSurveys: [] as Survey[],
+                searchSurveysCount: 0,
+            },
             loadSurveys: async () => {
-                const responseSurveys = await api.surveys.list()
-                return responseSurveys.results
+                const response = await api.surveys.list()
+                return {
+                    ...values.data,
+                    surveys: response.results,
+                    surveysCount: response.count,
+                }
+            },
+            loadNextPage: async () => {
+                const offset = values.data.surveys.length
+                const response = await api.surveys.list({
+                    limit: SURVEY_PAGE_SIZE,
+                    offset,
+                })
+
+                return {
+                    ...values.data,
+                    surveys: [...values.data.surveys, ...response.results],
+                    surveysCount: response.count,
+                }
+            },
+            loadSearchResults: async () => {
+                const trimmedSearchTerm = values.searchTerm?.trim() || ''
+                if (trimmedSearchTerm === '') {
+                    return {
+                        ...values.data,
+                        searchSurveys: [],
+                        searchSurveysCount: 0,
+                    }
+                }
+
+                // Only do backend search if we have more total items than the page size
+                if (values.data.surveysCount <= SURVEY_PAGE_SIZE) {
+                    return values.data
+                }
+
+                const response = await api.surveys.list({
+                    limit: SURVEY_PAGE_SIZE,
+                    search: trimmedSearchTerm,
+                })
+
+                return {
+                    ...values.data,
+                    searchSurveys: response?.results || [],
+                    searchSurveysCount: response?.count || 0,
+                }
+            },
+            loadNextSearchPage: async () => {
+                const offset = values.data.searchSurveys.length
+                const response = await api.surveys.list({
```

Comment:
should we also check here similar to L98-L101?

Reply:
i don't think so, as by definition, this is already for other pages. if there's no other pages, it'll be caught by the `hasNextSearchPage` and won't be available anyway